engine/src/object.cpp

```diff
@@ -5719,21 +5755,8 @@ void MCDeletedObjectsOnObjectDeleted(MCObject *p_object)
         return;
     
     // Unreference the pool.
-    t_pool -> references -= 1;
     p_object -> setdeletedobjectpool(nil);
-    
-    // Loop through any defunct pools.
-    while(t_pool -> defunct)
-    {
-        MCDeletedObjectPool *t_this_pool;
-        t_this_pool = t_pool;
-        t_pool = t_pool -> parent;
-        
-        if (t_this_pool -> references == 0)
-        {
-            MCDeletedObjectPoolDestroy(t_this_pool);
-        }
-    }
+	t_pool = MCDeletedObjectPoolGetNonDefunct(MCDeletedObjectPoolRelease(t_pool));
```

Comment:
Rather than making Release return a value, and then having another method wrapping it, it would be clearer to have:
```
/* Release the given object pool, and return the nearest non-defunct ancestor.
 * Any intermediate, unreferenced, defunct pools are destroyed. */
static MCDeletedObjectPool *MCDeletedObjectPoolUnwind(MCDeletedObjectPool *p_pool);
```